### DISCUSSION TITLE
Add UV_INDEX to the Weather Forecast data model

### DISCUSSION THREAD

**Discussion by jeeftor** [home-assistant/architecture#924] [Entity Models] [answered]
I would propose we add `uv_index` to the weather forecast data model. OpenWeather provides this data as `uvi`. The HACS integration for Tempest Weatherflow provides `uv_index`. 

For those living in hotter climates or higher altitudes this is important to track to know when to keep in or out of the sun.

Core integration APIs that support UVIndex:

- OpenWeather API
- Accuweather API
- [IPMA](https://www.home-assistant.io/integrations/ipma/)

**Comment by MartinHjelmare** [comment]
I think two integrations is too few to warrant a change to the entity model

Additionally, how we handle forecasts in Home Assistant may change with the recent addition of service return results.

**Reply by jeeftor** [reply]
What are service return results? Sounds interesting... 

Also whats a threshold ... I haven't looked that deep but it sounds like IPMA makes it 3 :)

**Reply by frenck** [reply]
I guess that still makes two, as custom integrations are not part of our considerations for model changes.

**Reply by gjohansson-ST** [reply]
These integration can also provide uv
- Tomorrow.io
- Met
- Open-Meteo

**Comment by dgomes** [comment]
[IPMA](https://www.home-assistant.io/integrations/ipma/) would also benefit (currently it's not exposed)

**Comment by frenck** [comment]
What would be the unit of measurement for these?

**Reply by jeeftor** [reply]
UV index is just an ~integer~ float.  It doesn't really have units. per/say

https://en.wikipedia.org/wiki/Ultraviolet_index

>The ultraviolet index, or UV index, is an international standard measurement of the strength of the [sunburn](https://en.wikipedia.org/wiki/Sunburn)-producing [ultraviolet](https://en.wikipedia.org/wiki/Ultraviolet) (UV) [radiation](https://en.wikipedia.org/wiki/Radiation) at a particular place and time. It is primarily used in daily and hourly forecasts aimed at the general public. The UV index is designed as an open-ended [linear scale](https://en.wikipedia.org/wiki/Linear_scale), directly proportional to the intensity of UV radiation, and adjusting for wavelength based on what causes [human skin](https://en.wikipedia.org/wiki/Human_skin) to sunburn.

**Reply by jeeftor** [reply]
https://www.who.int/news-room/questions-and-answers/item/radiation-the-ultraviolet-(uv)-index

>The UVI is a measure of the level of UV radiation.
The values of the index range from zero upward - the higher the UVI, the greater the potential for damage to the skin and eye, and the less time it takes for harm to occur.
The UVI is an important vehicle to alert people about the need to use sun protection.
A marked increase in the incidence of skin cancer in fair-skinned populations worldwide is strongly associated with excessive UV radiation exposure from the sun and possibly artificial sources such as sunbeds. Current evidence indicates that personal habits in relation to sun exposure constitute the most important individual risk factor for UV radiation damage.

> The UV Index is an important vehicle to raise public awareness of the risks of excessive exposure to UV radiation, and to alert people about the need to adopt protective measures. As part of an international effort, the UV index was developed by WHO, the United Nations Environment Programme, the World Meteorological Organization and the International Commission on Non-Ionizing Radiation Protection. Encouraging people to reduce their sun exposure can decrease harmful health effects and significantly reduce health care costs.

> The INTERSUN programme promotes the harmonized use of the UVI, and advises governments to employ this educational tool in their health promotion programmes. WHO encourages dissemination channels such as the media and tourism industry to publish the UVI forecast and promote sun protection messages.

**Reply by jeeftor** [reply]
On a personal level I have automations that alert me if the index is too high and I go make my children put on sunscreen...

**Reply by frenck** [reply]
Ok, so we have two Core integrations that could use this, right? OpenWeather and IPMA, both are using this particular index?

Reason I ask is that I'm afraid it would generally be smacked down to low/high and things like that, see also: <https://en.wikipedia.org/wiki/Ultraviolet_index#Index_usage>

**Reply by dgomes** [reply]
Same index

**Comment by jeeftor** [comment]
https://developer.accuweather.com/accuweather-forecast-api/apis/get/forecasts/v1/hourly/12hour/%7BlocationKey%7D

It appears accuweather has hourly uv index forecasts in the API as well... 

<img width="766" alt="image" src="https://github.com/home-assistant/architecture/assets/6491743/72889eb7-e250-4b5a-b702-21d13a162da3">

**Reply by jeeftor** [reply]
Tomorrow.io has support on some level:

<img width="861" alt="image" src="https://github.com/home-assistant/architecture/assets/6491743/093a988c-93e0-4c22-9d7c-051d8f4a1b82">

https://docs.tomorrow.io/reference/data-layers-core

**Comment by cyclops1982** [comment]
With https://github.com/home-assistant/core/pull/96195 i tried to do this for the [Met Office](https://www.home-assistant.io/integrations/metoffice/) integration too. The Met Office integration does have sensors for the UV index for the day, just not on an hourly basis. My aim would be to create a graph like mentioned [here](https://ukhsa.blog.gov.uk/2016/04/18/nine-things-you-need-to-know-about-uv-ultraviolet-radiation/) in point 6 and [here](https://www.rivm.nl/zonkracht)

**Reply by cyclops1982** [reply]
BTW, the use case for me is very similar to @jeeftor - knowing when to put sun cream on the kids.

**Comment by frenck** [comment, marked as answer]
We discussed this, and agree this one is OK to add to the entity model.

**Reply by gjohansson-ST** [reply]
PR made: https://github.com/home-assistant/core/pull/96951
